compiler/rustc_middle/src/ty/fast_reject.rs

```diff
@@ -76,10 +92,14 @@ pub fn simplify_type(
             _ => Some(MarkerTraitObjectSimplifiedType),
         },
         ty::Ref(_, ty, _) => {
-            // since we introduce auto-refs during method lookup, we
-            // just treat &T and T as equivalent from the point of
-            // view of possibly unifying
-            simplify_type(tcx, ty, can_simplify_params)
+            if strip_references == StripReferences::Yes {
+                // since we introduce auto-refs during method lookup, we
+                // just treat &T and T as equivalent from the point of
+                // view of possibly unifying
```

Comment:
that comment seems kind of wrong to me :thinking: 

I expect that this talks about https://github.com/rust-lang/rust/blob/ee86f96ba176f598d64dc9f3bb7e074d5b8b86b6/compiler/rustc_trait_selection/src/traits/select/mod.rs#L1935-L1938

but considering that `fast_reject_trait_refs` is only used in `match_impl` which uses https://github.com/rust-lang/rust/blob/ee86f96ba176f598d64dc9f3bb7e074d5b8b86b6/compiler/rustc_trait_selection/src/traits/select/mod.rs#L1904-L1908

if `fast_reject` returns false, it seems to me like ignoring references in `fast_reject` is causing us to sometimes do less work but doesn't actually interact with auto-refs in any way.

Am I missing something here? :thinking:

Reply:
Hmm, I think this code was initially *written* to speed up method dispatch. I think you are correct that this comment is outdated.

Reply:
updated the comment